src/adapters/postgres/user-adapter.ts

```diff
@@ -464,22 +490,34 @@ export class PostgresUserService implements IServicelocator {
   }
 
 
-  async createUserInDatabase(request: any, userCreateDto: UserCreateDto) {
+  async createUserInDatabase(request: any, userCreateDto: UserCreateDto, response: Response, apiId: string) {
     const user = new User()
     user.username = userCreateDto?.username
     user.name = userCreateDto?.name
     user.email = userCreateDto?.email
-    user.mobile = Number(userCreateDto?.mobile) || null,
-      user.createdBy = userCreateDto?.createdBy
+    user.mobile = userCreateDto?.mobile
+    user.dob = userCreateDto?.dob
+    user.createdBy = userCreateDto?.createdBy
     user.updatedBy = userCreateDto?.updatedBy
     user.userId = userCreateDto?.userId,
       user.state = userCreateDto?.state,
       user.district = userCreateDto?.district,
       user.address = userCreateDto?.address,
       user.pincode = userCreateDto?.pincode
 
+    if (userCreateDto?.email) {
+      user.encryptedEmail = encrypt(user.email)
+      user.email = maskPiiData('email', user.email)
+    }
+
+    if (userCreateDto?.mobile) {
+      user.encryptedMobile = encrypt(user.mobile)
+      user.mobile = maskPiiData('mobile', user.mobile)
+    }
+
     if (userCreateDto?.dob) {
-      user.dob = new Date(userCreateDto.dob);
+      user.encryptedDob = encrypt(user.dob)
+      user.dob = maskPiiData('dob', user.dob);
```

Comment:
user.dob = maskPiiData('dob', user.dob);
      this should be  
     user.dob = maskPiiData('date', user.dob);